libs/backend/src/routes/user/[username]/activity/index.ts

```diff
@@ -0,0 +1,40 @@
+import Puzzle from "@/models/puzzle/puzzle.js";
+import Submission from "@/models/submission/submission.js";
+import User from "@/models/user/user.js";
+import { FastifyInstance } from "fastify";
+import { userEntitySchema } from "types";
+
+export default async function userByUsernameActivityRoutes(fastify: FastifyInstance) {
+	fastify.get("/", async (request, reply) => {
+		const parseResult = userEntitySchema.pick({ username: true }).safeParse(request.params);
+
+		if (!parseResult.success) {
```

Comment:
could make a type-guard type of function, `isUsername`, not sure how often it's be used tho, so if this would be the only place that this logic is ever used, might not be needed and better to leave it here.

But I actually think, that there might be a function that does exactly this already!